src/java.base/share/classes/sun/security/pkcs/PKCS7.java

```diff
@@ -305,7 +313,7 @@ private void parseSignedData(DerValue val)
          * check if certificates (implicit tag) are provided
          * (certificates are OPTIONAL)
          */
-        if ((byte)(dis.peekByte()) == (byte)0xA0) {
+        if (dis.seeOptionalContextSpecific(0)) {
```

Comment:
The new call did not check bit 6. We should either enhance the seeOptionalXXX method or add another one to ensure that bit 6 has the expected value as well.

Reply:
You mean the constructed bit? The getSet() call in this if block will fail if it's not set. I wonder what difference it will make here. Suppose the bit 6 is not set but it's still [0], that means someone has stuffed a non-constructed data type here. In fact, since it's still [0], I think it should be better parsed inside this block and not somewhere else.